config/src/main/java/com/typesafe/config/ConfigMemorySize.java

```diff
@@ -0,0 +1,49 @@
+/**
+ *   Copyright (C) 2015 Typesafe Inc. <http://typesafe.com>
+ */
+package com.typesafe.config;
+
+/**
+ * An immutable class representing an amount of memory.  Use
+ * static factory methods such as {@link
+ * ConfigMemorySize#ofBytes(long)} to create instances.
+ */
+public final class ConfigMemorySize {
+    private final long bytes;
+
+    private ConfigMemorySize(long bytes) {
+        if (bytes < 0)
+            throw new IllegalArgumentException("Attempt to construct ConfigMemorySize with negative number: " + bytes);
+        this.bytes = bytes;
+    }
+
+    /** Constructs a ConfigMemorySize representing the given
```

Comment:
this is kind of a weird javadoc. normally i'd see it written as:

    /**
     * Constructs a ConfigMemorySize representing the given
     * number of bytes.
     */